sessions/forecasting-concepts.qmd

```diff
@@ -4,413 +4,231 @@ order: 7
-  I0 = 1,
-  obs = long_onset_df$onsets,
-  gen_time_max = length(gen_time_pmf),
-  gen_time_pmf = gen_time_pmf,
-  ip_max = length(ip_pmf) - 1,
-  ip_pmf = ip_pmf,
-  h = 0
-)
-
-
-rw_long <- nfidd_sample(mod,
-  data = long_data,
-  adapt_delta = 0.95,
-  init = \() list(init_R = 1, rw_sd = 0.01)
-)
+The `fable` package has a number of common forecasting models ready for you to use. In this session we're just going to try one without thinking too much about what it is and why it might be good or bad. We're going to start with an Auto-Regressive Integrated Moving Average (ARIMA) model. Some of the core ideas of ARIMA models have been in use for 100 years, although they have been around in their current form for the last 50.[@tsay2000time] We recommend [the FPP3 ARIMA Chapter](https://otexts.com/fpp3/arima.html) as a good applied introduction to ARIMA modeling.
+
+Here is the code to fit an ARIMA model that models the an observation $y_t$ as a function of its previous two observations. As with other regression models in R, you can inspect some details about the fit.
+```{r fit-first-model}
+fit <- flu_data |>
+  model( arima200 = ARIMA(wili ~ pdq(2,0,0)) )
+report(fit)
 ```
 
-We first need to extract the forecast and estimated reproduction numbers.
+And now that we've fitted a model, we can use it to make a forecast. We'll start by making a forecast for `h=25` weeks into the future. (Although usually for epidemiological forecasts, this would be a bit further than most models could make accurate or well-calibrated forecasts.)
 
-```{r extract-Rt}
-forecast_r <- rw_forecast |>
-  gather_draws(R[day]) |>
-  ungroup() |> 
-  mutate(type = "forecast")
+```{r make-a-forecast}
+first_forecast <- forecast(fit, h=25) 
 
-long_r <- rw_long |>
-  gather_draws(R[day]) |>
-  ungroup() |>
-  mutate(type = "estimate")
+first_forecast |>
+  autoplot(flu_data |> filter(epiweek >= as.Date("2016-09-01"))) +
+  labs(title = "WILI, US level",
+       y="% of outpatient visits due to ILI")
 ```
 
-We can now plot the forecast and estimated reproduction numbers.
-
-```{r plot-Rt}
-forecast_r |>
-  bind_rows(long_r) |>
-  filter(.draw %in% sample(.draw, 100)) |>
-  ggplot(aes(x = day)) +
-  geom_vline(xintercept = cutoff, linetype = "dashed") +
-  geom_hline(yintercept = 1, linetype = "dashed") +
-  geom_line(
-    aes(y = .value, group = interaction(.draw, type), color = type),
-    alpha = 0.1
-  ) +
-  labs(
-    title = "Estimated R",
-    subtitle = "Estimated over whole time series (red), and forecast (blue)"
-  ) +
-  guides(colour = guide_legend(override.aes = list(alpha = 1)))
-```
+Note that in the above plot we filtered the observed data so that the resulting plot wasn't dominated by all of the historical seasons of data. The plot shows just last season's observations and "forecasts" for 25 weeks of the 2017/2018 season. Note that the predictions are probabilistic in that they show a predicted mean (the solid line) and an 80% and 95% prediction interval. 
```

Comment:
we have found it easier to manage if all sentences go on a new line. Happy not to continue doing that if that feels clunky